### PR TITLE
[jasmine] Add type definitions for Jasmine 3.6

### LINKED ISSUE #46368: [jasmine] Update @types/jasmine to reflect API changes in jasmine-core 3.6.0
At least, the typing for the new toHaveBeenCalledOnceWith matcher is missing

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: 
@borisyankov @theodorejb @davidparsson @gmoothart @lukas-zech-software @Engineer2B @cyungmann @Roaders @devoto13 @fdim @pe8ter @kolodny @stephenfarrar @zvirja @djungowski @chivesrs

### PR DESCRIPTION
**Pull request by ienzam**
Fixes #46368

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://jasmine.github.io/api/3.6/global.html#setSpecProperty
https://jasmine.github.io/api/3.6/global.html#setSuiteProperty
https://jasmine.github.io/api/3.6/jasmine.html#.addCustomObjectFormatter
https://jasmine.github.io/api/3.6/MatchersUtil.html#pp
https://jasmine.github.io/api/3.6/Env.html#setSpecProperty
https://jasmine.github.io/api/3.6/Env.html#setSuiteProperty
https://jasmine.github.io/api/3.6/matchers.html#toHaveBeenCalledOnceWith
https://jasmine.github.io/api/3.6/matchers.html#toHaveSize

- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -382,6 +407,16 @@ declare namespace jasmine {
...
         contains<T>(haystack: ArrayLike<T> | string, needle: any, customTesters?: ReadonlyArray<CustomEqualityTester>): boolean;
         buildFailureMessage(matcherName: string, isNot: boolean, actual: any, ...expected: any[]): string;
+
+        /**
+         * Formats a value for use in matcher failure messages and similar
+         * contexts, taking into account the current set of custom value
+         * formatters.
+         * @since 3.6.0
+         * @param value - The value to pretty-print
+         * @return - The pretty-printed value
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:417]: Drop the - before The?

**Review by chivesrs — APPROVED** [review]
This looks ok but I'd prefer any over unknown to be consistent with current types.

File: types/jasmine/index.d.ts (hunk 1 above)
Comment (ienzam) [reply types/jasmine/index.d.ts]: Thanks for the review :D 

> This looks ok but I'd prefer any over unknown to be consistent with current types.

Agreed with the consistency argument. In general, we should move to using more unknown, especially in defining function types to force type check. Leaving as is if you don't have strong preference.

> Drop the - before The?

Done. Updated for the whole file.

**Review by chivesrs — APPROVED** [review]

**Comment by ienzam** [comment]
Ready to merge

**Pull request merged**